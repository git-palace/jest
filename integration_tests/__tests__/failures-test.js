/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

'use strict';

const {extractSummary} = require('../utils');
const path = require('path');
const runJest = require('../runJest');
const skipOnWindows = require('skipOnWindows');

const dir = path.resolve(__dirname, '../failures');

skipOnWindows.suite();

test('throwing not Error objects', () => {
  let stderr;
  stderr = runJest(dir, ['throw-number-test.js']).stderr;
  expect(extractSummary(stderr)).toMatchSnapshot();
  stderr = runJest(dir, ['throw-string-test.js']).stderr;
  expect(extractSummary(stderr)).toMatchSnapshot();
  stderr = runJest(dir, ['throw-object-test.js']).stderr;
  expect(extractSummary(stderr)).toMatchSnapshot();
  stderr = runJest(dir, ['assertion-count-test.js']).stderr;
  expect(extractSummary(stderr)).toMatchSnapshot();
});
