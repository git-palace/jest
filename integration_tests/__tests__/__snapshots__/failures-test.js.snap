exports[`test throwing not Error objects 1`] = `
" FAIL  __tests__/throw-number-test.js
  ● Test suite failed to run

    Expected an Error, but \"1\" was thrown
    

1 test suite failed, 0 tests passed (0 total in 1 test suite, <TIME>)
"
`;

exports[`test throwing not Error objects 2`] = `
" FAIL  __tests__/throw-string-test.js
  ● Test suite failed to run

    Error
      banana

1 test suite failed, 0 tests passed (0 total in 1 test suite, <TIME>)
"
`;

exports[`test throwing not Error objects 3`] = `
" FAIL  __tests__/throw-object-test.js
  ● Test suite failed to run

ERROR: No message was provided

1 test suite failed, 0 tests passed (0 total in 1 test suite, <TIME>)
"
`;
